discord_bot2.py

```diff
@@ -163,8 +164,105 @@ async def on_ready(self):
         logging.info(self.user.name)
         logging.info(self.user.id)
         logging.info('------')
+        channels = self.get_all_channels()
+        for channel in channels:
+            if channel.name == 'control-duck':
+                await channel.send('Duck online')
 
-    async def on_message(self, message):
+    async def restart(self, message):
+        """
+        Restart the bot
+        :param message: The message that triggered the restart
+        """
+        await message.channel.send(f'Restart requested.')
+        os.chdir(Path(__file__).parent)
+        await self.execute_command('git fetch', message.channel)
+        await self.execute_command('git reset --hard', message.channel)
+        await self.execute_command('git clean -f', message.channel)
+        await self.execute_command('git pull --rebase=false', message.channel)
+        await self.execute_command('poetry install', message.channel)
+        await message.channel.send(f'Restarting.')
+        subprocess.Popen(["bash", "restart.sh"])
+        return
+
+    async def say_in_channel(self, channel, text, prefix="", suffix=""):
+        """
+        Say something in a channel
+        :param channel: The channel to say something in
+        :param output: The message to say, long messages will be split into multiple messages
+        """
+        # Async methods like for loops better than while loops
+        # Split output into 1800 character chunks
+        for i in range(len(text) // 1800):
+            await channel.send(f'{prefix}{text[:1800]}{suffix}')
+            text = text[1800:]
+
+        await channel.send(f'{prefix}{text}{suffix}')
```

Comment:
Isn't this logic already present somewhere? Why rewrite it?

Reply:
Sorry about that, I didn't realize that's what `parse_blocks` did. Since `send` currently uses `parse_blocks` but takes a thread, I think I'll make a similar `send_in_channel`.